modelforge/tests/test_dataset.py

```diff
@@ -1208,3 +1208,118 @@ def test_shifting_center_of_mass_to_origin(prep_temp_dir):
 
         assert torch.allclose(pairs.r_ij, pairs_ns.r_ij, atol=1e-4)
         assert torch.allclose(pairs.d_ij, pairs_ns.d_ij, atol=1e-4)
+
+
+@pytest.mark.parametrize("dataset_name", _ImplementedDatasets.get_all_dataset_names())
+def test_element_filter(dataset_name, prep_temp_dir):
+    local_cache_dir = str(prep_temp_dir) + "/data_test"
+
+    atomic_number = np.array(
+        [
+            [1],
+            [2],
+            [3],
+            [4],
+            [5],
+            [6],
+            [7],
+            [8],
+            [9],
+            [10],
+            [11],
+            [12],
+        ]
+    )
+
+    # positive tests
+
+    # Case 1
+    data = _ImplementedDatasets.get_dataset_class(
+        dataset_name,
+    )(
+        version_select="nc_1000_v0",
+        local_cache_dir=local_cache_dir,
+        element_filter=[(1,)],
+    )
+    assert data._satisfy_element_filter(atomic_number)
+
+    # Case 2
+    data = _ImplementedDatasets.get_dataset_class(
+        dataset_name,
+    )(
+        version_select="nc_1000_v0",
+        local_cache_dir=local_cache_dir,
+        element_filter=[(1, 2, 3)],
+    )
+    assert data._satisfy_element_filter(atomic_number)
+
+    # Case 3
+    data = _ImplementedDatasets.get_dataset_class(
+        dataset_name,
+    )(
+        version_select="nc_1000_v0",
+        local_cache_dir=local_cache_dir,
+        element_filter=[(1, 2), (3, 4), (5, 6)],
+    )
+    assert data._satisfy_element_filter(atomic_number)
+
+    # Case 4
+    data = _ImplementedDatasets.get_dataset_class(
+        dataset_name,
+    )(
+        version_select="nc_1000_v0",
+        local_cache_dir=local_cache_dir,
+        element_filter=[(1, 2), (3, -15)],
```

Comment:
you need to define a tests in which you are swapping the order of the filter around 

Design a filter that should evaluate to true (1,2) and then one that should be false (-3) and swap the order to ensure that you get a True out of it. 

[[1,2], [-3]] 
[[-3], [1,2]]

The bug in the code using "is" to compare was not being caught by these tests